### PR TITLE
Build wheels for all OS using CI

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Partially addresses #85 by using cibuildwheels to create more wheels for installing apexpy.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Needs CI to test

**Test Configuration**:
* Operating system: GitHub Actions
* Python version number: N/A
* Compiler with version number: N/A
* Any details about your local setup that are relevant (e.g., apexpy version/branch)

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``Changelog.rst``, summarising the changes
- [x] Add yourself to ``AUTHORS.rst`` and ``.zenodo.json``

### CODE REVIEW

**Comment by ljlamarche** [comment]
Changes look fine, but how should I test this?  Following the new instructions under maintenance?

**Comment by aburrell** [comment]
Yes, if the new maintenance instructions work for you then you've shown:
1) The instructions work for someone other than myself
2) The wheels exist.

You could also try installing using one of the wheels, if it's not hard.

**Comment by ljlamarche** [comment]
How do you choose which artifact to select?  Just the most recent?

**Comment by ljlamarche** [comment]
I couldn't get the zip file to download through my browser, but could get it using `wget` with the url specified by `location`.  It contains the following wheels:
```
Archive:  wheel_zip
  inflating: apexpy-1.1.0-cp310-cp310-linux_x86_64.whl  
  inflating: apexpy-1.1.0-cp310-cp310-macosx_10_16_x86_64.whl  
  inflating: apexpy-1.1.0-cp310-cp310-win_amd64.whl  
  inflating: apexpy-1.1.0-cp37-cp37m-linux_x86_64.whl  
  inflating: apexpy-1.1.0-cp37-cp37m-macosx_10_16_x86_64.whl  
  inflating: apexpy-1.1.0-cp37-cp37m-win_amd64.whl  
  inflating: apexpy-1.1.0-cp38-cp38-linux_x86_64.whl  
  inflating: apexpy-1.1.0-cp38-cp38-macosx_10_16_x86_64.whl  
  inflating: apexpy-1.1.0-cp38-cp38-win_amd64.whl  
  inflating: apexpy-1.1.0-cp39-cp39-linux_x86_64.whl  
  inflating: apexpy-1.1.0-cp39-cp39-macosx_10_16_x86_64.whl  
  inflating: apexpy-1.1.0-cp39-cp39-win_amd64.whl  
  inflating: numpy-1.21.6-cp37-cp37m-win_amd64.whl  
  inflating: numpy-1.23.5-cp310-cp310-win_amd64.whl  
  inflating: numpy-1.23.5-cp38-cp38-win_amd64.whl  
  inflating: numpy-1.23.5-cp39-cp39-win_amd64.whl  
```
I did manage to install the appropriate wheel for my computer and everything appears to work!

File: docs/maintenance.rst [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
+To download an artifact:
+
+1. If you don't have a GitHub Personal Access Token, follow
+   `these instructions <https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token>`_
+   to create one.
+2. Run ``curl -v -H "Authorization: token <GITHUB-ACCESS-TOKEN>" https://api.github.com/repos/aburrell/apexpy/actions/artifacts/<ARTIFACT-ID>/zip``, where
+   <ITEM> should be replaced with the appropriate item string.
+3. Copy the URL from the ``Location`` output produced by the previous command
+   into a browser, which will download a zip archive into your standard
+   download location.
```
Comment (aburrell) [inline docs/maintenance.rst:22]:
```suggestion
   download location. Alternatively (or if this doesn't work) you can use `wget` to retrieve the archive.
```

**Comment by aburrell** [comment]
I updated the download instructions based on your experience.  If you're happy, go ahead and approve the pull request :)

**Review by ljlamarche — APPROVED** [review]
Looks good to me!

**Pull request merged**